### PR TITLE
Do not send a stale state channel

### PR DESCRIPTION
**Pull request by macpie**

### CODE REVIEW

**Comment by macpie** [comment]
I have also removed `handlers` they are unused here anyway

**Review by michaeldjeffrey — APPROVED** [review]

File: src/state_channel/blockchain_state_channels_server.erl [hunk 1]
```diff
@@ -74,21 +74,26 @@ get_all() ->
...
+                fun({SC, SCState, Pid}) ->
+                    ID = blockchain_state_channel_v1:id(SC),
+                    try blockchain_state_channels_worker:get(Pid) of
+                        UpToDateSC ->
+                            {ID, {UpToDateSC, SCState, Pid}}
+                        
+                    catch _C:_E ->
+                        Name = blockchain_utils:addr2name(ID),
+                        lager:error("failed to get sc ~p(~p) ~p/~p", [Name, Pid, _C, _E]),
+                        lager:error("~p", [recon:info(Pid)]),
```
Comment (vihu) [inline src/state_channel/blockchain_state_channels_server.erl:91]: Is this okay? I guess depends on the occurrence?

**Review by vihu — APPROVED** [review]

File: src/state_channel/blockchain_state_channels_server.erl (hunk 1 above)
Comment (jeffgrunewald) [reply src/state_channel/blockchain_state_channels_server.erl:91]: calling recon:info/1 or returning undefined where we used to return that nested tuple?

Comment (vihu) [reply src/state_channel/blockchain_state_channels_server.erl:91]: Just error spam, the undefined later would get filtered out.

**Pull request merged**